sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -291,9 +292,10 @@ static int32_t _az_http_client_curl_upload_read_callback(
 
   int32_t data_length = az_span_length(*mut_span);
   int32_t to_copy = (data_length < curl_size) ? data_length : curl_size;
-  AZ_RETURN_IF_FAILED(
-      az_span_append(*mut_span, az_span_init((uint8_t *)ptr, to_copy, to_copy), mut_span));
-
+  memcpy(ptr, mut_span->_internal.ptr, to_copy);
```

Comment:
NO code but the span code itself should be directly accessing _internal members of a span structure. Can this be modified to avoid doing this?

Reply:
I agree.  The old code didn't transfer over 1-1 due to the mutating buffer being initialized when we map to it.  The old code took advantage of behaviors in the mutating span which are no longer present.  I'll play with it and see if I can make it better.  At a minimum I can update this code to use the access helpers